tests/integration/instagram-search-test.js

```diff
@@ -8,12 +10,11 @@ var App;
 module('Integration: Instagram Search', {
   setup: function() {
     App = startApp();
-
-    App.register('instagram-api-client:main', Ember.Object.extend({
-       recentMediaForTag: function(tag, options) {
-         return Ember.RSVP.resolve(yoloInstagramRecentMediaJson);
-       }
-     }));
+    App.__container__.lookup('service:instagram-api-client').reopen({
```

Comment:
Before, we would just register a stub for `instagram-api-client:main`. The new `Ember.Service` pattern automatically registers and injects the service you defined.

Since we don't use the initializer anymore, we can't tell our app: "don't register our service in `Ember.testing`" with the `if (!Ember.testing)` guard. Instead, we `lookup` the service, reopen it, and override the *methods* we want to stub.